source/states/PlayerAirState.hx

```diff
@@ -0,0 +1,94 @@
+package states;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.input.keyboard.FlxKey;
+
+class PlayerAirState extends PlayerState
+{
+  public function new()
+  {
+    super();
+
+    onGround = false;
+  }
+
+  private override function transition():Bool
+  {
+    if (isPlayerOnGround())
+    {
+      nextState = new PlayerGroundState();
+      return true;
+    }
+    return false;
+  }
+
+  public override function enter(object:FlxObject):Void
+  {
+      var player = cast(object, Player);
+      player.velocity.y = -400;
+  }
+
+  private override function action(object:FlxObject):Void
+  {
+    var player = cast(object, Player);
+    var horizontalMove:Int = pollForHorizontalMove();
+
+    if (isPlayerRunning())
+    {
+      player.xMaxSpeed = player.runSpeed;
+    }
+    else
+    {
+      player.xMaxSpeed = player.walkSpeed;
+    }
+
+    // If horizontalMove is -1, the Player should move left.
+    if (horizontalMove == -1)
+    {
+      if (player.velocity.x > 0)
+      {
+        player.acceleration.x = -player.xSlowdown + -player.xAccel;
+      }
+      else
+      {
+        player.acceleration.x = -player.xAccel;
+      }
+    }
+
+    // If horizontalMove is 1, the Player should move right.
+    else if (horizontalMove == 1)
+    {
+      if (player.velocity.x < 0)
+      {
+        player.acceleration.x = player.xSlowdown + player.xAccel;
+      }
+      else
+      {
+        player.acceleration.x = player.xAccel;
+      }
+    }
+
+    // Slow down if no direction held
```

Comment:
This comment is a bit misleading (and I'm pretty sure I'm the one that wrote it, sorry). The player doesn't slow down in the air if no direction is held, they just stop accelerating and maintain their current momentum.